src/coreclr/jit/emitxarch.h

```diff
@@ -414,7 +415,9 @@ static bool IsRexW1EvexInstruction(instruction ins);
 
 bool IsThreeOperandAVXInstruction(instruction ins)
 {
-    return (IsDstDstSrcAVXInstruction(ins) || IsDstSrcSrcAVXInstruction(ins));
+    return (IsDstDstSrcAVXInstruction(ins) || IsDstSrcSrcAVXInstruction(ins) ||
+            (ins == INS_vpcmpb || ins == INS_vpcmpw || ins == INS_vpcmpd || ins == INS_vpcmpq || ins == INS_vpcmpub ||
+             ins == INS_vpcmpuw || ins == INS_vpcmpud || ins == INS_vpcmpuq));
```

Comment:
Any reason these can't have the `IsDstDstSrcAvxInstruction` flag? Was the concern that they aren't `DstDstSrc`, but are rather `MskDstSrc`?

Reply:
I think that was the reason yes

Reply:
I think it would be good to introduce a new flag to represent this and potentially some mask flag `Is3OperandMask` that is a combination of the three, so we can check this much more cheaply.

Reply:
Actually, sorry. I looked at just the `bool emitter::IsThreeOperandInstructionMask(instruction ins) const` implementation....

I'd expect this function to just become:
```cpp
bool IsThreeOperandAVXInstruction(instruction ins)
{
    if (!UseSimdEncoding())
    {
        return false;
    }

    insFlags flags = CodeGenInterface::instInfo[ins];
    return (flags & INS_Flags_Is3OperandInstructionMask) != 0;
}
```

I'd then expect us to have:
```cpp
    INS_Flags_IsDstDstSrcAVXInstruction = 1ULL << 25,
    INS_Flags_IsDstSrcSrcAVXInstruction = 1ULL << 26,
    INS_Flags_Is3OperandInstruction     = 1ULL << 27, // Alternatively name this `INS_Flags_IsMskSrcSrcEvexInstruction`

    INS_Flags_Is3OperandInstructionMask = (INS_Flags_IsDstDstSrcAVXInstruction | INS_Flags_IsDstSrcSrcAVXInstruction | INS_Flags_Is3OperandInstruction),
```

That way we have the existing two flags and a new flag to cover the new concept, but we can have a single check to cover all three of them from the perspective of "this has 3 registers involved"